packages/gatsby-theme-docs/gatsby-config.js

```diff
@@ -71,6 +72,7 @@ module.exports = (themeOptions = {}) => {
         options: {
           name: 'internalConfigurationData',
           path: path.join(__dirname, `./src/data`),
+          ignore: pluginOptions.overrideTopMenuLinks ? ['**/top-*'] : [],
```

Comment:
This will basically ignore the `data/top*.yaml` files from the theme, thus allowing the files from the website `src/data/top*.yaml` to be solely used.